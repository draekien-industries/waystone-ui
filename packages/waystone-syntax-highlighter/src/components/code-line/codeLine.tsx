'use client';

import { useMemo } from 'react';
import { Box } from 'theme-ui';
import type { Node, StyleSheet } from '../../lib';
import { createStyleObject } from '../../utils';

export type CodeLineProps = {
  node: Node;
  stylesheet: StyleSheet;
};

export const CodeLine = ({ node, stylesheet }: CodeLineProps) => {
  const allStylesheetSelectors = useMemo(
    () =>
      Object.keys(stylesheet).reduce((classes, selector) => {
        selector.split('.').forEach((className) => {
          if (!classes.includes(className)) {
            classes.concat(className);
          }
        });

        return classes;
      }),
    [stylesheet]
  );

  if (node.type === 'text') {
    return <span>{node.value}</span>;
  }

  const { properties } = node;

  const startingClassName =
    properties.className && properties.className.includes('token')
      ? ['token']
      : [];

  const className =
    properties.className &&
    startingClassName.concat(
      properties.className.filter(
        (cname) => !allStylesheetSelectors.includes(cname)
      )
    );

  const props = {
    ...properties,
    className: className.join(' ') || undefined,
    style: createStyleObject({
      classNames: properties.className,
      stylesheet,
    }),
  };

  return (
    <Box as={node.tagName || 'span'} {...props}>
      {node.children.map((child, index) => (
        <CodeLine
          node={child}
          stylesheet={stylesheet}
          key={`code-line-${index.toString()}`}
        />
      ))}
    </Box>
  );
};
