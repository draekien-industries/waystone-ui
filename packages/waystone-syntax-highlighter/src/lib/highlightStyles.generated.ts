//
// This file has been auto-generated by the 'pnpm generate:styles' task
//


export const SYNTAX_HIGHLIGHT_STYLES = [
  'a11yDark',
  'a11yLight',
  'agate',
  'anOldHope',
  'androidstudio',
  'arduinoLight',
  'arta',
  'ascetic',
  'atomOneDark',
  'atomOneDarkReasonable',
  'atomOneLight',
  'brownPaper',
  'codepenEmbed',
  'colorBrewer',
  'dark',
  'defaultStyle',
  'devibeans',
  'docco',
  'far',
  'felipec',
  'foundation',
  'github',
  'githubDark',
  'githubDarkDimmed',
  'gml',
  'googlecode',
  'gradientDark',
  'gradientLight',
  'grayscale',
  'hybrid',
  'idea',
  'intellijLight',
  'irBlack',
  'isblEditorDark',
  'isblEditorLight',
  'kimbieDark',
  'kimbieLight',
  'lightfair',
  'lioshi',
  'magula',
  'monoBlue',
  'monokai',
  'monokaiSublime',
  'nightOwl',
  'nnfxDark',
  'nnfxLight',
  'nord',
  'obsidian',
  'pandaSyntaxDark',
  'pandaSyntaxLight',
  'paraisoDark',
  'paraisoLight',
  'pojoaque',
  'purebasic',
  'qtcreatorDark',
  'qtcreatorLight',
  'rainbow',
  'routeros',
  'schoolBook',
  'shadesOfPurple',
  'srcery',
  'stackoverflowDark',
  'stackoverflowLight',
  'sunburst',
  'tokyoNightDark',
  'tokyoNightLight',
  'tomorrowNightBlue',
  'tomorrowNightBright',
  'vs',
  'vs2015',
  'xcode',
  'xt256'
] as const;

export type SyntaxHighlightStyle = (typeof SYNTAX_HIGHLIGHT_STYLES)[number];

export default SYNTAX_HIGHLIGHT_STYLES;
