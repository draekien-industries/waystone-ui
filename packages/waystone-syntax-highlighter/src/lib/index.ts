export * from './highlightStyles.generated';
export * from './supportedLanguages.generated';
export * from './styles';
