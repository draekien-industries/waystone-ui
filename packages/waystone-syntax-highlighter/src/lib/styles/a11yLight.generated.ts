export const a11yLight = {
  're code.hljs': {
    display: 'block',
    overflowX: 'auto',
    padding: '1em',
  },
  'ode.hljs': {
    padding: '3px 5px',
  },
  hljs: {
    background: '#fefefe',
    color: '#545454',
  },
  'hljs-comment': {
    color: '#696969',
  },
  'hljs-quote': {
    color: '#696969',
  },
  'hljs-deletion': {
    color: '#d91e18',
  },
  'hljs-name': {
    color: '#d91e18',
  },
  'hljs-regexp': {
    color: '#d91e18',
  },
  'hljs-selector-class': {
    color: '#d91e18',
  },
  'hljs-selector-id': {
    color: '#d91e18',
  },
  'hljs-tag': {
    color: '#d91e18',
  },
  'hljs-template-variable': {
    color: '#d91e18',
  },
  'hljs-variable': {
    color: '#d91e18',
  },
  'hljs-attribute': {
    color: '#aa5d00',
  },
  'hljs-built_in': {
    color: '#aa5d00',
  },
  'hljs-link': {
    color: '#aa5d00',
  },
  'hljs-literal': {
    color: '#aa5d00',
  },
  'hljs-meta': {
    color: '#aa5d00',
  },
  'hljs-number': {
    color: '#aa5d00',
  },
  'hljs-params': {
    color: '#aa5d00',
  },
  'hljs-type': {
    color: '#aa5d00',
  },
  'hljs-addition': {
    color: 'green',
  },
  'hljs-bullet': {
    color: 'green',
  },
  'hljs-string': {
    color: 'green',
  },
  'hljs-symbol': {
    color: 'green',
  },
  'hljs-section': {
    color: '#007faa',
  },
  'hljs-title': {
    color: '#007faa',
  },
  'hljs-keyword': {
    color: '#7928a1',
  },
  'hljs-selector-tag': {
    color: '#7928a1',
  },
  'hljs-emphasis': {
    fontStyle: 'italic',
  },
  'hljs-strong': {
    fontWeight: '700',
  },
};
