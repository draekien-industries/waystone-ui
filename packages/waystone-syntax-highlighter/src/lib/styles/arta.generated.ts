export const arta = {
  're code.hljs': {
    display: 'block',
    overflowX: 'auto',
    padding: '1em',
  },
  'ode.hljs': {
    padding: '3px 5px',
  },
  hljs: {
    background: '#222',
    color: '#aaa',
  },
  'hljs-subst': {
    color: '#aaa',
  },
  'hljs-section': {
    color: '#fff',
    fontWeight: '700',
  },
  'hljs-comment': {
    color: '#444',
  },
  'hljs-meta': {
    color: '#444',
  },
  'hljs-quote': {
    color: '#444',
  },
  'hljs-bullet': {
    color: '#fc3',
  },
  'hljs-regexp': {
    color: '#fc3',
  },
  'hljs-string': {
    color: '#fc3',
  },
  'hljs-symbol': {
    color: '#fc3',
  },
  'hljs-addition': {
    color: '#0c6',
  },
  'hljs-number': {
    color: '#0c6',
  },
  'hljs-attribute': {
    color: '#32aaee',
  },
  'hljs-built_in': {
    color: '#32aaee',
  },
  'hljs-link': {
    color: '#32aaee',
  },
  'hljs-literal': {
    color: '#32aaee',
  },
  'hljs-template-variable': {
    color: '#32aaee',
  },
  'hljs-type': {
    color: '#32aaee',
  },
  'hljs-keyword': {
    color: '#64a',
  },
  'hljs-name': {
    color: '#64a',
  },
  'hljs-selector-class': {
    color: '#64a',
  },
  'hljs-selector-id': {
    color: '#64a',
  },
  'hljs-selector-tag': {
    color: '#64a',
  },
  'hljs-deletion': {
    color: '#b16',
  },
  'hljs-template-tag': {
    color: '#b16',
  },
  'hljs-title': {
    color: '#b16',
  },
  'hljs-variable': {
    color: '#b16',
  },
  'hljs-doctag': {
    fontWeight: '700',
  },
  'hljs-strong': {
    fontWeight: '700',
  },
  'hljs-emphasis': {
    fontStyle: 'italic',
  },
};
