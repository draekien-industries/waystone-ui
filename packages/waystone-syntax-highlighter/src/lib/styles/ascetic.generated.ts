export const ascetic = {
  're code.hljs': {
    display: 'block',
    overflowX: 'auto',
    padding: '1em',
  },
  'ode.hljs': {
    padding: '3px 5px',
  },
  hljs: {
    background: '#fff',
    color: '#000',
  },
  'hljs-addition': {
    color: '#888',
  },
  'hljs-attribute': {
    color: '#888',
  },
  'hljs-bullet': {
    color: '#888',
  },
  'hljs-link': {
    color: '#888',
  },
  'hljs-section': {
    color: '#888',
    fontWeight: '700',
  },
  'hljs-string': {
    color: '#888',
  },
  'hljs-symbol': {
    color: '#888',
  },
  'hljs-template-variable': {
    color: '#888',
  },
  'hljs-variable': {
    color: '#888',
  },
  'hljs-comment': {
    color: '#ccc',
  },
  'hljs-deletion': {
    color: '#ccc',
  },
  'hljs-meta': {
    color: '#ccc',
  },
  'hljs-quote': {
    color: '#ccc',
  },
  'hljs-keyword': {
    fontWeight: '700',
  },
  'hljs-name': {
    fontWeight: '700',
  },
  'hljs-selector-tag': {
    fontWeight: '700',
  },
  'hljs-strong': {
    fontWeight: '700',
  },
  'hljs-type': {
    fontWeight: '700',
  },
  'hljs-emphasis': {
    fontStyle: 'italic',
  },
};
