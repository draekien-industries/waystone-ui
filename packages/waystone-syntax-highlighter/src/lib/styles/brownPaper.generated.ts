export const brownPaper = {
  're code.hljs': {
    display: 'block',
    overflowX: 'auto',
    padding: '1em',
  },
  'ode.hljs': {
    padding: '3px 5px',
  },
  hljs: {
    color: '#363c69',
    background: 'url(./brown-papersq.png) #b7a68e',
  },
  'hljs-keyword': {
    color: '#059',
    fontWeight: '700',
  },
  'hljs-literal': {
    color: '#059',
    fontWeight: '700',
  },
  'hljs-selector-tag': {
    color: '#059',
    fontWeight: '700',
  },
  'hljs-addition': {
    color: '#2c009f',
  },
  'hljs-attribute': {
    color: '#2c009f',
  },
  'hljs-built_in': {
    color: '#2c009f',
  },
  'hljs-bullet': {
    color: '#2c009f',
  },
  'hljs-link': {
    color: '#2c009f',
  },
  'hljs-name': {
    color: '#2c009f',
    fontWeight: '700',
  },
  'hljs-section': {
    color: '#2c009f',
    fontWeight: '700',
  },
  'hljs-string': {
    color: '#2c009f',
  },
  'hljs-symbol': {
    color: '#2c009f',
  },
  'hljs-template-tag': {
    color: '#2c009f',
  },
  'hljs-template-variable': {
    color: '#2c009f',
  },
  'hljs-title': {
    color: '#2c009f',
    fontWeight: '700',
  },
  'hljs-type': {
    color: '#2c009f',
    fontWeight: '700',
  },
  'hljs-variable': {
    color: '#2c009f',
  },
  'hljs-comment': {
    color: '#802022',
  },
  'hljs-deletion': {
    color: '#802022',
  },
  'hljs-meta': {
    color: '#802022',
  },
  'hljs-quote': {
    color: '#802022',
  },
  'hljs-doctag': {
    fontWeight: '700',
  },
  'hljs-strong': {
    fontWeight: '700',
  },
  'hljs-emphasis': {
    fontStyle: 'italic',
  },
};
