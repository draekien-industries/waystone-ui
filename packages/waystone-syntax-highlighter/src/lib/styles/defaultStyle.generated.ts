export const defaultStyle = {
  're code.hljs': {
    display: 'block',
    overflowX: 'auto',
    padding: '1em',
  },
  'ode.hljs': {
    padding: '3px 5px',
  },
  hljs: {
    background: '#f3f3f3',
    color: '#444',
  },
  'hljs-comment': {
    color: '#697070',
  },
  'hljs-punctuation': {
    color: '#444a',
  },
  'hljs-tag': {
    color: '#444a',
  },
  'hljs-tag .hljs-attr': {
    color: '#444',
  },
  'hljs-tag .hljs-name': {
    color: '#444',
  },
  'hljs-attribute': {
    fontWeight: '700',
  },
  'hljs-doctag': {
    fontWeight: '700',
  },
  'hljs-keyword': {
    fontWeight: '700',
  },
  'hljs-meta .hljs-keyword': {
    fontWeight: '700',
  },
  'hljs-name': {
    fontWeight: '700',
  },
  'hljs-selector-tag': {
    fontWeight: '700',
  },
  'hljs-deletion': {
    color: '#800',
  },
  'hljs-number': {
    color: '#800',
  },
  'hljs-quote': {
    color: '#800',
  },
  'hljs-selector-class': {
    color: '#800',
  },
  'hljs-selector-id': {
    color: '#800',
  },
  'hljs-string': {
    color: '#800',
  },
  'hljs-template-tag': {
    color: '#800',
  },
  'hljs-type': {
    color: '#800',
  },
  'hljs-section': {
    color: '#800',
    fontWeight: '700',
  },
  'hljs-title': {
    color: '#800',
    fontWeight: '700',
  },
  'hljs-link': {
    color: '#ab5656',
  },
  'hljs-operator': {
    color: '#ab5656',
  },
  'hljs-regexp': {
    color: '#ab5656',
  },
  'hljs-selector-attr': {
    color: '#ab5656',
  },
  'hljs-selector-pseudo': {
    color: '#ab5656',
  },
  'hljs-symbol': {
    color: '#ab5656',
  },
  'hljs-template-variable': {
    color: '#ab5656',
  },
  'hljs-variable': {
    color: '#ab5656',
  },
  'hljs-literal': {
    color: '#695',
  },
  'hljs-addition': {
    color: '#397300',
  },
  'hljs-built_in': {
    color: '#397300',
  },
  'hljs-bullet': {
    color: '#397300',
  },
  'hljs-code': {
    color: '#397300',
  },
  'hljs-meta': {
    color: '#1f7199',
  },
  'hljs-meta .hljs-string': {
    color: '#38a',
  },
  'hljs-emphasis': {
    fontStyle: 'italic',
  },
  'hljs-strong': {
    fontWeight: '700',
  },
};
