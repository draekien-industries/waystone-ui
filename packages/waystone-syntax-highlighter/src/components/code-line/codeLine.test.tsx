import { render } from '@waystone/testing-library';
import { CodeLine } from './codeLine';
import { githubDark } from '../../lib/styles';
import { Node } from '../../lib';

describe('Components: CodeLine', () => {
  it('matches snapshot', () => {
    const node: Node = {
      type: 'element',
      tagName: 'span',
      properties: {
        className: ['re code'],
      },
      children: [
        {
          type: 'text',
          value: 'text value',
        },
      ],
    };

    const { container } = render(
      <CodeLine node={node} stylesheet={githubDark} />
    );

    expect(container).toMatchSnapshot();
  });
});
