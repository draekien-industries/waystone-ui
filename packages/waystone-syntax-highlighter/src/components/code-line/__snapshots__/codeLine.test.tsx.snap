// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: CodeLine matches snapshot 1`] = `
<div>
  <span
    class="css-g89m8z-Box"
  >
    <span>
      text value
    </span>
  </span>
</div>
`;
